### PR TITLE
feat: app/play store rating

### LINKED ISSUE #6: add a "do you like the app? you can review"

### PR DESCRIPTION
**Pull request by abhiShandy**
fixes #6 

- adds an entry in the settings screen to rate the app, this is a fallback
- we need to discuss the scenarios where the user will be asked to rate the app

### CODE REVIEW

**Comment by daviroo** [comment]
@abhiShandy I think we would want to ask the user for a rating after they have sent their first successful payment and perhaps after receiving 2-3 payments (not suitable to ask on first receipt since this might be very soon after they first open the app).  What do you think?

**Comment by nicolasburtey** [comment]
I would start to ask for a review when they receive their second payment (or any upcoming payment if they have already have received 2)

**Comment by agbegin** [comment]
When to ask user to review/discussion with Abhi: 

1. Why reviews? Be more findable in the app store, improve discoverability. Increase conversion of the app listing. 
2. Any existing feedback mechanisms? no, other than support link -- Muun has good experience, feedback button in the top corner. They got back to ask for more details (within a day) 
3. Idea: new item in menu called "Provide feedback", has two items in it: Suggest a feature or improvement (creates github issue/support ticket), and "Rate on the app store" 
4. Prompt flow: 
    - Enjoying this app? Yes / No
    - If yes: Great! Would you like to leave a review on the app (Play) store to help others find it? (yes/no)
    - If no: Thank you for the feedback, would you like to suggest an improvement or feature request 
5. Triggers for prompt: 
    - Successfully receive 2nd payment 
6. Dismiss/delay for when they deny leaving a review or providing feedback
    - First time denying, turn off for 1 mo 
    - Second time denying, turn off for 6 mo 
7. Backend changes required: 
    - Record when user has either left feedback or reviewed the app (rather, clicked out to review the app) 
    - Save timestamp to user account for dismissing the feedback prompt
8. Data capture: 
    - Capture user feedback about whether they are enjoying app or not (@jireva @bodymindarts to advise on how/where) 
    - Capture other analytics related to user interaction (% who leave feedback or review) 
    - Mutations on the graphQL side may be needed – e.g. if user uninstalls and comes back 
9. Decide on a nice time to release
    - @agbegin to coordinate on ideal timing

**Comment by sandipndev** [comment]
hey @abhiShandy, I am helping accelerate the work here about ratings - looking at what's done here.

**Comment by sandipndev** [comment]
@agbegin and @nicolasburtey, iiuc, what we want is:

**Comment by sandipndev** [comment]
Proposals:

- [ ] Settings screen to have "Leave Feedback" instead of "Rate Us" which opens a Modal, which has the options:
    - Rate us on (App/Play) Store -- Opens the Rating dialog box (which also asks for store review at the end)
    - Star us on GitHub -- Opens https://github.com/GaloyMoney/galoy
    - Suggest an Improvement -- `sendFeedback()`

- [ ] After receiving their first payment, start a custom prompt flow:
    - [ ] Enjoying this app? Yes / No
      - [ ] If yes: Great! Would you like to rate us? Yes / No
        - [ ] If yes: Open Rate Dialog box (which also asks for store review at the end)
    - [ ] If no: Thank you for the feedback, would you like to suggest an improvement or feature request? Yes / No
        - [ ] If yes: `sendFeedback()`

`sendFeedback()` => Opens a custom reviews modal where users can write text and submit. The description can be "Tell us what you love about our app, or what we could be doing better."

This does mean that we need a graphql query to submit the user feedback.

**Comment by sandipndev** [comment]
image

File: app/screens/settings-screen/settings-screen.tsx [hunk 1]
```diff
@@ -139,11 +141,23 @@ export const SettingsScreen: React.FC = () => {
...
 
+  const rateUs = () => {
+    Rate.rate(ratingOptions, (success, errorMessage) => {
+      if (success) {
+        // this technically only tells us if the user successfully went to the Review Page.
+        // whether they actually did anything, we do not know.
+      }
+      if (errorMessage) {
+        // errorMessage comes from the native code. Useful for debugging, but probably not for users to view
+        console.error(`Rate.rate() error: ${errorMessage}`)
```
Comment (Extheoisah) [inline app/screens/settings-screen/settings-screen.tsx:156]: we should probably also log the error in crashlytics so we have eyes there.
`crashlytics().recordError(err)`

Comment (sandipndev) [reply app/screens/settings-screen/settings-screen.tsx:156]: thanks, adding this.

**Review by sandipndev — APPROVED** [review]

**Pull request merged**